src/snowflake/snowpark/_internal/analyzer/expression.py

```diff
@@ -118,6 +118,9 @@ def cumulative_node_complexity(self) -> Dict[PlanNodeCategory, int]:
     def cumulative_node_complexity(self, value: Dict[PlanNodeCategory, int]):
         self._cumulative_node_complexity = value
 
+    def reset_cumulative_node_complexity(self) -> None:
```

Comment:
who is calling this? didn't see it is called in this pr

Reply:
we do it for logical_plan node but not for expression. I'll remove this